cpp/src/arrow/filesystem/azurefs.h

```diff
@@ -92,30 +92,30 @@ struct ARROW_EXPORT AzureOptions {
 
  private:
   enum class CredentialKind {
+    kDefault,
     kAnonymous,
-    kTokenCredential,
-    kStorageSharedKeyCredential,
-  } credential_kind_ = CredentialKind::kAnonymous;
+    kStorageSharedKey,
+    kClientSecret,
+    kManagedIdentity,
+    kWorkloadIdentity,
```

Comment:
I don't think we really need to change this. I would rather just keep `kTokenCredential` to cover all the credentials that are based on https://github.com/Azure/azure-sdk-for-cpp/blob/e5e675440b44ace7d7a9e7bc303f877c06b59ea5/sdk/core/azure-core/inc/azure/core/credentials/credentials.hpp#L68

Reply:
We need this to support `Equals`. Think of it as runtime type-information that describes what concrete implementation of `TokenCredential` is being used.

Reply:
Besides that, we would need to make a distinction between `kDefault` and all the others `kToken` at least (to support the implicit default behavior). It's clearer if we then make a distinction on all of them.

Reply:
I think I would have done it differently but I don't feel strongly.